source/game/kart/KartMove.cc

```diff
@@ -311,6 +615,14 @@ bool KartMove::canWheelie() const {
     return false;
 }
 
+bool KartMove::canHop() const {
+    if (state()->isHopStart() && state()->isTouchingGround()) {
+        return true;
+    } else {
+        return false;
+    }
+}
```

Comment:
```suggestion
    if (!state()->isHopStart() || !state()->isTouchingGround()) {
        return false;
    }
    
    return true;
}
```
This is closer to the base game for when we need to implement the `HIT_ITEM_OR_OBJ` flag.